docs/tutorial/mysql.md

```diff
@@ -0,0 +1,99 @@
+
+USE pets;
+
+CREATE TABLE mytable (
+    name            varchar(80),
+    pet             varchar(80)
+);
+
+INSERT INTO mytable VALUES ('Mary', 'dog'), ('John', 'cat'), ('Robert', 'bird');
+```
+
+## Add username and password to your local app secrets
+
+Your local Streamlit app will read secrets from a file `.streamlit/secrets.toml` in your app's root directory. Create this file if it doesn't exist yet and add the database name, user, and password of your MySQL server as shown below:
+
+```python
+# .streamlit/secrets.toml
+
+[mysql]
+host = "localhost"
+port = 3306
+database = "xxx"
+user = "xxx"
+password = "xxx"
+```
+
+```eval_rst
+.. important:: Add this file to ``.gitignore`` and don't commit it to your Github repo!
+```
+
+## Copy your app secrets to the cloud
+
+As the `secrets.toml` file above is not committed to Github, you need to pass its content to your deployed app (on Streamlit sharing or Streamlit for Teams) separately. Go to the [app dashboard](https://share.streamlit.io/) and in the app's dropdown menu, click on **Edit Secrets**. Copy the content of `secrets.toml` into the text area. More information in [Secrets Management](../deploy_streamlit_app.html#secrets-management).
+
+![](../media/databases/postgresql-1.png)
+
+## Add mysql-connector-python to your requirements file
+
+Add the [mysql-connector-python](https://github.com/mysql/mysql-connector-python) package to your `requirements.txt` file, preferably pinning its version (replace `x.x.x` with the version you want installed):
+
+```
+# requirements.txt
+mysql-connector-python==x.x.x
+```
+
+## Write your Streamlit app
+
+Copy the code below to your Streamlit app and run it. Make sure to adapt `query` to use the name of your table.
+
+```python
+# streamlit_app.py
+
+import streamlit as st
+import mysql.connector
+
+# Initialize connection.
+# Uses st.cache to only run once.
+@st.cache(allow_output_mutation=True, hash_funcs={"_thread.RLock": lambda _: None})
+def init_connection():
+    return mysql.connector.connect(**st.secrets["mysql"])
+
+conn = init_connection()
+
+# Perform query.
+# Uses st.cache to only rerun when the query changes or after 10 min.
+@st.cache(ttl=600)
+def run_query(query):
+    with conn.cursor() as cur:
+        cur.execute(query)
+        return cur.fetchall()
+
+rows = run_query("SELECT * from mytable;")
+
+# Print results.
+for row in rows:
+    st.write(f"{row[0]} has a :{row[1]}:")
+```
+
+See `st.cache` above? Without it, Streamlit would run the query every time the app reruns (e.g. on a widget interaction). With `st.cache`, it only runs when the query changes or after 10 minutes (that's what `ttl` is for). Watch out: If your database updates more frequently, you should adapt `ttl` or remove caching so viewers always see the latest data. Read more about caching [here](../caching.md).
+
+If everything worked out (and you used the example table we created above), your app should look like this:
+
+![](../media/databases/postgresql-2.png)
```

Comment:
Yes, it is intentional. I noticed yesterday that all the database tutorials [share one image](https://docs.streamlit.io/en/stable/_images/big-query-10.png) in common, and five tutorials [share another image](https://docs.streamlit.io/en/stable/_images/public-gsheet-4.png). Rather than being a single image file referenced by multiple tutorials, I had unnecessarily duplicated images for each tutorial. This is redundant and adds to bloat.

For this PR, I decided not to create two additional duplicates and instead link to images used in the `postgres` tutorial. 

The plan was to create a follow-up PR that cleaned up / de-duplicated these database tutorial images. Should I include these changes in this MySQL PR?